source/ranges.tex

```diff
@@ -4477,15 +4490,33 @@
 \item
 Otherwise, if \tcode{T} models
 \libconcept{random_access_range} and \libconcept{sized_range}
-and is
+and is a specialization of
+\tcode{span}\iref{views.span},
+\tcode{basic_string_view}\iref{string.view}, or
+\tcode{ranges::subrange}\iref{range.subrange},
+then
+\tcode{U(ranges::begin(E),
+ranges::be\-gin(E) + std::min<D>(ranges::distance(E), F))},
+except that \tcode{E} is evaluated only once,
+where \tcode{U} is a type determined as follows:
+
 \begin{itemize}
-\item a specialization of \tcode{span}\iref{views.span} where \tcode{T::extent == dynamic_extent},
-\item a specialization of \tcode{basic_string_view}\iref{string.view},
-\item a specialization of \tcode{ranges::iota_view}\iref{range.iota.view}, or
-\item a specialization of \tcode{ranges::subrange}\iref{range.subrange},
+\item if \tcode{T} is a specialization of \tcode{span},
+then \tcode{U} is \tcode{span<typename T::element_type>};
+\item otherwise, if \tcode{T} is a specialization of \tcode{basic_string_view},
+then \tcode{U} is \tcode{T};
+\item otherwise, \tcode{T} is a specialization of \tcode{ranges::subrange}, and
+\tcode{U} is \tcode{ranges::subrange<iterator_t<T>>};
 \end{itemize}
-then \tcode{T(ranges::begin(E), ranges::begin(E) + min<D>(ranges::size(E), F))},
-except that \tcode{E} is evaluated only once.
+
+\item
+otherwise, if \tcode{T} is
+a specialization of \tcode{ranges::iota_view}\iref{range.iota.view}
+that models \libconcept{random_access_range} and \libconcept{sized_range},
+then
+\tcode{ranges::iota_view(*ranges::begin(E),
+*(ranges::begin(E) + std::\linebreak{}min<D>(ranges::distance(E), F)))},
+except that \tcode{E} and \tcode{F} are each evaluated only once;
```

Comment:
`F` is only mentioned once in the expression, so clearly it's evaluated only once and we don't need to say it again. I think this was a copy-paste error of mine, sorry :(

```suggestion
except that \tcode{E} is evaluated only once;
```